components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -1639,6 +1639,8 @@ else if ("subcontext".equalsIgnoreCase(searchType)) {
                     criteria=APIConstants.API_OVERVIEW_CONTEXT;
                 }else if (searchType.equalsIgnoreCase("Description")) {
                     criteria=APIConstants.API_OVERVIEW_DESCRIPTION;
+                } else if (searchType.equalsIgnoreCase("Tag")) {
```

Comment:
This should be changed as APIConstants.API_TAG.equalsIgnoreCase(searchType) to avoid an NPE if searchType becomes null